src/misc_early.rs

```diff
@@ -113,4 +125,13 @@ impl EarlyLintPass for MiscEarly {
             }
         }
     }
+
+    fn check_expr(&mut self, cx: &EarlyContext, expr: &Expr) {
+        if let ExprKind::Call(ref path, ref args) = expr.node {
+            if let ExprKind::Closure(_, _, ref block) = path.node {
```

Comment:
To expand on this a bit, the AST contains the syntax tree. As such, things like `(((((a)))))` are part of it. However, this gets removed during "desugaring" in the HIR (which LateLintPass uses). The HIR looks similar to the AST but has a lot of syntax sugar removed -- for example a for loop will be rewritten as a `loop` and a `match`, `if let` becomes a match, etc.